nipype/interfaces/dtitk/registration.py

```diff
@@ -1,16 +1,28 @@
-from ..base import TraitedSpec, CommandLineInputSpec, traits, isdefined
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""DTITK registration interfaces
+
+    Change directory to provide relative paths for doctests
+    >>> import os
+    >>> filepath = os.path.dirname( os.path.realpath( __file__ ) )
+    >>> datadir = os.path.realpath(os.path.join(filepath, '../../testing/data'))
+    >>> os.chdir(datadir)
+"""
+
+from ..base import TraitedSpec, CommandLineInputSpec, traits, isdefined, File
 from ...utils.filemanip import fname_presuffix
 import os
 from .base import CommandLineDtitk
 
 
 class RigidInputSpec(CommandLineInputSpec):
-    fixed_file = traits.Str(desc="fixed diffusion tensor image",
-                            exists=True, mandatory=True,
-                            position=0, argstr="%s")
-    moving_file = traits.Str(desc="diffusion tensor image path", exists=True,
-                             mandatory=True, position=1, argstr="%s")
-    similarity_metric = traits.Enum('EDS', 'GDS', 'DDS', 'NMI', exists=True,
+    fixed_file = File(desc="fixed diffusion tensor image",
+                      exists=True, mandatory=True,
+                      position=0, argstr="%s")
+    moving_file = File(desc="diffusion tensor image path", exists=True,
+                       mandatory=True, position=1, argstr="%s", copyfile=False)
+    similarity_metric = traits.Enum('EDS', 'GDS', 'DDS', 'NMI',
                                     mandatory=True, position=2, argstr="%s",
                                     desc="similarity metric")
     samplingX = traits.Float(mandatory=True, position=3, argstr="%s",
```

Comment:
@satra: some of these functions require parameters to be set on the command line (no defaults), but we provide default values in the input spec (I matched defaults to the example from the DTITK tutorial), fulfilling this requirement. I am redoing the input specs such that usedefault=True and mandatory=False. In general, does that sound like a good idea, or is it better to make the user specify?

Reply:
if it's a common default value across use cases then setting a default value is fine. however, if it's a required argument, `mandatory` should always be set to `True` even if a default value is used. also `mandatory=False` should not be used.